api/src/pcapi/connectors/big_query/queries/pro_no_bookings_since_40_days_ago.py

```diff
@@ -0,0 +1,22 @@
+from pcapi import settings
+
+from .base import BaseQuery
+from .base import RowIterator
+from .models import ProEmailModel
+
+
+class NoBookingsProEmail(BaseQuery):
+    raw_query = f"""
+        select
+            distinct venue_booking_email
+        from
+            `{settings.BIG_QUERY_NOTIFICATIONS_TABLE_BASENAME}.marketing_pro_email_last_booking_40_days_ago`
+        where
+            cast(execution_date as date) = current_date()
+    """
+
+    model = ProEmailModel
+
+    def execute(self, page_size: int = 1_000) -> RowIterator:
+        for row in super().execute(page_size):
+            yield ProEmailModel(venue_booking_email=row.venue_booking_email)
```

Comment:
on passerait pas un paramètre de page pour récupérer la page suivante ?

Reply:
pas besoin, `super().execute(page_size)` requête toutes les pages (`BigQueryRow` de taille `page_size`), les aplatit et renvoie un itérateur
test :
```python
staging❯ import pydantic
    ...: from pcapi import settings
    ...: 
    ...: from pcapi.connectors.big_query.queries.base import BaseQuery
    ...: 
    ...: class IdModel(pydantic.BaseModel):
    ...:     offer_id: int
    ...: 
    ...: class FavoritesNotBooked(BaseQuery):
    ...:     raw_query = f"""
    ...:         SELECT
    ...:             offer_id
    ...:         FROM
    ...:             `{settings.BIG_QUERY_NOTIFICATIONS_TABLE_BASENAME}.favorites_not_booked`
    ...:         LIMIT 10
    ...:     """
    ...: 
    ...:     model = IdModel
    ...: 
    ...: rows = FavoritesNotBooked().execute(page_size=3)
    ...: print(", ".join((str(row.offer_id) for row in rows)))
8104900, 6652358, 41452373, 7674312, 83375093, 83375093, 6652358, 77249794, 6652358, 20962545
```
mais me pencher sur la question m'a permis de trouver une implem plus simple

Reply:
Toop